### PR TITLE
mark slow local tests

### PR DESCRIPTION
**Pull request by willingc**
# References and relevant issues
Partially addresses #7382

# Description
This PR marks the 10 slowest tests on my local machine. This allows me to include
or exclude when I run tests locally.

This could be an initial proof of concept step that could be used to identify some slow
tests on CI runs and treat them differently than the rest of the test suite. Perhaps exclude on some test matrix runs or parallelize the slow tests.

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -359,7 +359,7 @@ filterwarnings = [
   "ignore:There is no current event loop:DeprecationWarning:",
   "ignore:(?s).*Pyarrow will become a required dependency of pandas:DeprecationWarning",  # pandas pyarrow (pandas<3.0),
   # TODO: remove once xarray is updated to avoid this warning
-  # https://github.com/pydata/xarray/blame/b1f3fea467f9387ed35c221205a70524f4caa18b/pyproject.toml#L333-L334
+  # https://github.com/pydata/xarray/blame/b1f3fea467ff9387ed35c221205a70524f4caa18b/pyproject.toml#L333-L334
```
Comment (psobolewskiPhD) [inline pyproject.toml]: The link doesn't work; this change is a typo I think -- I spot an extra `f`.

Comment (willingc) [reply pyproject.toml]: Oops that was a stray keypress. Fixed.

File: pyproject.toml [hunk 2]
```diff
@@ -373,6 +373,7 @@ markers = [
     "disable_qtimer_start: Disable timer start in this Test",
     "disable_qanimation_start: Disable animation start in this Test",
     "enable_console: Don't mock the IPython console (in QtConsole) in this Test",
+    "slow: mark a test as slow",
```
Comment (psobolewskiPhD) [inline pyproject.toml:378]:
```suggestion
    # mark slow tests, so they can be skipped using: pytest -m "not slow"
    "slow: mark a test as slow",
```

I had to look up mark usage, so maybe this is helpful to add here?

**Review by psobolewskiPhD — APPROVED** [review]
Thanks! I think this is useful!
I added a suggestion of a comment to document usage.

**Comment by Czaki** [comment]
For search slow test (also these that are slow because of multiple entry in parametrization) you may want to use https://carreau.github.io/pytest-json-report-viewer/ 

It allows getting plot based on json reports from test run.

**Comment by willingc** [comment]
> For search slow test (also these that are slow because of multiple entry in parametrization) you may want to use https://carreau.github.io/pytest-json-report-viewer/

Very cool. I didn't know that Matthias made this. Thanks for mentioning @Czaki.

File: napari/utils/_tests/test_io.py [hunk 3]
```diff
@@ -72,6 +73,7 @@ def test_imsave_float(tmp_path, image_file):
         assert not image_file_path.is_file()
 
 
+@pytest.mark.slow
```
Comment (Czaki) [inline napari/utils/_tests/test_io.py]:
```suggestion
```

As the #7384 is approved

**Pull request merged**